internal/pkg/pluginmanager/pbdownloader.go

```diff
@@ -0,0 +1,119 @@
+// 1. get plugins *.so
+// 2. show progress bar on console
+
+package pluginmanager
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/cheggaaa/pb"
+)
+
+type PbDownloadClient struct {
+	*http.Client
+}
+
+func NewPbDownloadClient() *PbDownloadClient {
+	dClient := PbDownloadClient{}
+	dClient.Client = http.DefaultClient
+	return &dClient
+}
+
+// download from release assets
+func (pd *PbDownloadClient) download(pluginsDir, pluginFilename, version string) error {
+
+	err := createPathIfNotExists(pluginsDir)
+	if err != nil {
+		return err
+	}
+
+	downloadURL := fmt.Sprintf("%s/v%s/%s", defaultReleaseUrl, version, pluginFilename)
+	tmpName := pluginFilename + ".tmp"
+
+	pd.Timeout = time.Second * 60 * 60
+	resp, err := pd.Get(downloadURL)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		log.Printf("[INFO] downloading: [%s]", pluginFilename)
+		log.Print("\n")
```

Comment:
Why not just use `log.Printf("[INFO] downloading: [%s]**\n**", pluginFilename)`

Reply:
done